_msprimemodule.c

```diff
@@ -2016,11 +2016,7 @@ MutationModel_init(MutationModel *self, PyObject *args, PyObject *kwds)
             "transition matrix must be a square matrix with num_alleles rows");
         goto out;
     }
-    if (num_alleles != PyObject_Size(py_transition_matrix)) {
```

Comment:
Oops, this was left over from a previous version. You can just delete the assert.